File: geqiangbao_20201117_CommandPattern_01/geqiangbao_20201117_CommandPattern_01/day1_Command/computer/MacShutDownCommand.m
//
//  MacShutDownCommand.m
//  geqiangbao_20201117_CommandPattern_01
//
//  Created by xiaobao on 2020/11/17.
//

#import "MacShutDownCommand.h"

@interface MacShutDownCommand ()
@property (nonatomic,strong) MacComputer *computer;
@end


@implementation MacShutDownCommand
-(instancetype)initWith:(MacComputer*)computer{
    self = [super init];
    if (self) {
        self.computer = computer;
    }
    return self;
}

-(void)execute{
    [self.computer shutDown];
}


@end

File: geqiangbao_20201117_CommandPattern_swift/geqiangbao_20201117_CommandPattern_swift/computer/MacShutDownCommand.swift
//
//  MacShutDownCommand.swift
//  geqiangbao_20201117_CommandPattern_swift
//
//  Created by xiaobao on 2020/11/17.
//

import Cocoa

class MacShutDownCommand: NSObject {
    private var computer:MacComputer
    init(_ computer:MacComputer) {
        self.computer = computer;
    }
    func execute() {
        self.computer.shutDown()
    }
}
